File: examples/src/main/scala/SimpleSQLContext.scala
import com.provectus.lymph.LymphJob
import org.apache.spark.sql.SQLContext

object SimpleSQLContext extends LymphJob {
  /** Contains implementation of spark job with ordinary [[org.apache.spark.sql.SQLContext]]
    * Abstract method must be overridden
    *
    * @param context    spark sql context
    * @param parameters user parameters
    * @return result of the job
    */
  override def doStuff(context: SQLContext, parameters: Map[String, Any]): Map[String, Any] = {
    val df = context.read.json(parameters("file").asInstanceOf[String])
    df.registerTempTable("people")
    Map("result" -> context.sql("SELECT AVG(age) AS avg_age FROM people").collect())
  }
}

File: examples/src/main/python/simple_sql_context.py
from mist.mist_job import *

class SimpleSQLContext(MistJob, WithSQLSupport):

    def do_stuff(self, parameters):
        val = parameters.values()
        patch = val.head()

        df = self.sql_context.read.json(patch)
        df.printSchema()
        df.registerTempTable("people")

        df.show()
        df2 = self.sql_context.sql("SELECT AVG(age) AS avg_age FROM people")
        df2.show()

        result = df2.toJSON().first()

        return result
